src/charm.py

```diff
@@ -355,10 +351,23 @@ def _holistic_handler(self, event: EventBase) -> None:
             return
 
         self._workload_service.prepare_dir(path=LOG_DIR)
+        if self.unit.is_leader():
+            self.peer_data.prepare()
 
         # Install the certificates in various event scenarios
-        certs = TLSCertificates.load(self.certificate_transfer_requirer)
-        self._workload_service.push_ca_certs(certs.ca_bundle)
+        self._workload_service.push_ca_certs(self._ca_bundle)
+
+        try:
+            self._pebble_service.plan(self._pebble_layer)
+        except PebbleError:
+            logger.error(
+                f"Failed to plan pebble layer, please check the {WORKLOAD_CONTAINER} container logs"
+            )
+            raise
+
+    @cached_property
```

Comment:
Just a **NONBLOCKING** question: why do we declare a `cached_property` here? Does it bring a computation benefit?

Reply:
The property will be accessed from:
1) https://github.com/canonical/identity-platform-admin-ui-operator/pull/18/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR358
2) https://github.com/canonical/identity-platform-admin-ui-operator/pull/18/files#diff-51246e53255db77c9edad496f074aa1bdbf8dbdc11f89a02040115c9ab4fa7f0R70

This means that it will be called multiple times per hook. I thought that it would be beneficial not to access the databag multiple times, afaik the databag is cached so it should be that expensive. I can switch it back if you think that it does not add much.

Reply:
no worries. we can keep it like this way :)